src/ORM/DataObject.php

```diff
@@ -3724,6 +3724,51 @@ public function onAfterBuild()
         $this->extend('onAfterBuild');
     }
 
+    private function getDatabaseBackedField(string $fieldPath): ?string
+    {
+        $component = $this;
+        $fieldParts = [];
+
+        foreach ($parts = explode('.', $fieldPath ?? '') as $nextPart) {
+            if (!$component) {
+                return null;
+            }
+            $fieldParts[] = $nextPart;
+            
+            if ($component instanceof Relation || $component instanceof DataList) {
+                if ($component->hasMethod($nextPart)) {
+                    // If the next part is a method, we don't have a database-backed field.
+                    return null;
+                }
+                // The next part could either be a field, or another relation
+                $singleton = DataObject::singleton($component->dataClass());
+                if ($singleton->dbObject($nextPart)) {
+                    // If the next part is a DBField, we've found the database-backed field.
+                    break;
+                }
+                $component = $component->relation($nextPart);
+                array_shift($parts);
```

Comment:
What is the `array_shift` call doing there? Why do we need that? The tests pass without it, so either a test scenario is missing, or this isn't needed.
(the one below _is_ needed for the tests to pass, though)

Reply:
Updated. New test cases were added.